docs/examples/tunnel-with-reverse-proxy/README.md

```diff
@@ -0,0 +1,161 @@
+```
+3. Replace all placeholder values formatted `<like-this>`.
+   - in `manifests/cloudflare-operator`
+      - `<email-address>`: the email address associated with the cloudflare zone.
+      - `<domain>`: the domain associated with the cloudflare zone you are managing.
+      - `<secret-name>`: the name of the secret containing your cloudflare credentials.
+      - `<account-id>`: the account ID associated with the cloudflare zone you are managing.
+   - in `manifests/hello/ingress.yaml`
+     - `<domain>`: the domain associated with the cloudflare zone you are managing.
+4. deploy the cluster tunnel and tunnelbinding
+```shell
+kubectl apply -f manifests/cloudflare-operator/
+```
+5. deploy the reverse proxy
+```shell
+kustomize build --enable-helm manifests/ingress-nginx | kubectl apply -f - 
+```
+6. deploy the example application
+```shell
+kubectl apply -f manifests/hello/
+```
+7. You should now be able to access the service on `hello.<domain>`
+
+## Extending this example
+
+Now that we are using an arbitrary reverse proxy for routing traffic into the cluster, we can configure it as if we were using a cloud load balancer.
+This means you can use whatever constructs are available to your reverse proxy implementation.
+
+### Custom SSO
+
+One can configure SSO with a custom provider.
+This example uses ingress-nginx and authelia
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: <app>
+  annotations:
+    cert-manager.io/cluster-issuer: "letsencrypt-prod"
+    # in order to direct cloudflare tunnel to ingress-nginx, we have a single tunnelBinding resource in ingress-nginx.
+    # this is what controls ingress to ingress-nginx and allows us to put 2fa infront of apps that don't support it natively.
+    nginx.ingress.kubernetes.io/rewrite-target: /
+    nginx.ingress.kubernetes.io/auth-method: "GET"
+    nginx.ingress.kubernetes.io/auth-url: "http://authelia.authelia.svc.cluster.local:8080/api/authz/auth-request"
+    nginx.ingress.kubernetes.io/auth-signin: "https://authelia.<domain>?rm=$request_method"
+    nginx.ingress.kubernetes.io/auth-response-headers: "Remote-User,Remote-Name,Remote-Groups,Remote-Email"
+--- SNIP ---
+```
+
+### Using cloudflared for routing as well as ingress-nginx
+
+Using this approach does not prohibit additional routes.
+This is useful if you are migrating to/from one approach, or have different requirements for different endpoints.
+For example, you may want some endpoints to bypass your reverse proxy for some reason.
+
+```yaml
+apiVersion: networking.cfargotunnel.com/v1alpha1
+kind: TunnelBinding
+metadata:
+  name: authelia
+  namespace: authelia
+subjects:
+  - name: authelia # the fqdn authelia.<domain> maps to this service
+tunnelRef:
+  kind: ClusterTunnel
+  name: example-tunnel
+---
+apiVersion: networking.cfargotunnel.com/v1alpha1
+kind: TunnelBinding
+metadata:
+  name: ingress-nginx
+subjects:
+  - name: wildcard
+    spec:
+      fqdn: "*.<domain>"
```

Comment:
I'd be careful officially recommending a wildcard mixed with regular tunnel binding (like anthelia). Does this work seamlessly for you? I believe there was a sorting problem that was not fully fixed yet (sorting by routing entries instead of sorting by tunnel binding)

May be throw in a note?

Reply:
Yeah this has worked solidly for me for about 1.5 years 🤷

Reply:
What sort of note would you like me to write? I've not come across this issue

Reply:
I wonder if it has something to do with how your tunnel bindings are named that makes the wildcard be below the specific routes in the ConfigMap. Mind if I ask you to check your tunnel config map real quick to see if the wildcard is the last entry? If not, could you try accessing the one below the wildcard and see if it hits the ingress controller or not?

For reference, the top to bottom [mentioned in the doc](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/do-more-with-tunnels/local-management/configuration-file/#how-traffic-is-matched) is my concern. I would much rather we recommend replacing the catch all `fallbackTarget` (currently defaults to `http_status:404`) with the ingress service url.

Reply:
[created an issue](https://github.com/adyanth/cloudflare-operator/issues/147) and will document this behaviour

Reply:
have documented

Reply:
Looks like that repo might be private. Any concerns with my suggestion of using the fallbackTarget for the ingress service? It works better than the wild card by redirecting all unknown traffic to the ingress to decide.

Reply:
oops, it's public now :)

> Any concerns with my suggestion of using the fallbackTarget for the ingress service? It works better than the wild card by redirecting all unknown traffic to the ingress to decide

How does one configure the fields [in the tunnelBinding type](https://github.com/adyanth/cloudflare-operator/blob/fc07f48f8d2bc853cf5b8156b0e7ded8be0dc40d/api/v1alpha1/tunnelbinding_types.go#L35) with this approach?

Reply:
It is configured in the tunnel: https://github.com/adyanth/cloudflare-operator/blob/fc07f48f8d2bc853cf5b8156b0e7ded8be0dc40d/api/v1alpha1/tunnel_types.go#L125

Reply:
oh sure, but it prevents specifying protocol, fqdn, caPool, noTLSVerify, Http2Origin, Proxy* fields

Reply:
I personally find it useful using the fallback as `http_status:404` as it gives me an extra layer of observability into ingress

Reply:
That's interesting, and a valid point. Cool, I agree with documenting for now. Although, the name of the tunnelBinding is the one that needs to change, not the subject (as of now that is).

Reply:
Ah okay, will update